kai/reactive_codeplanner/task_manager/task_manager.py

```diff
@@ -93,21 +99,53 @@ def run_validators(self) -> list[Task]:
         logger.info("Running validators.")
         validation_tasks: list[Task] = []
 
-        for validator in self.validators:
+        def run_validator(validator):
             logger.debug("Running validator: %s", validator)
-            result = validator.run()
-            logger.debug("Validator result: %s", result)
-            if not result.passed:
-                validation_tasks.extend(result.errors)
-                logger.info(
-                    "Found %d new tasks from validator %s",
-                    len(result.errors),
+            try:
+                result = validator.run()
+                return validator, result
+            except Exception as e:
+                logger.error(
+                    "Validator %s failed to execute with an unhandled error: %s",
                     validator,
+                    e,
                 )
-                logger.debug("Validator %s found errors: %s", validator, result.errors)
+                return validator, None
+
+        with ThreadPoolExecutor() as executor:
+            # Submit all validators to the executor
+            future_to_validator = {
+                executor.submit(run_validator, v): v for v in self.validators
+            }
+
+            # Process results as they complete
+            for future in as_completed(future_to_validator):
```

Comment:
Should we have a timeout variable?

Reply:
Probably eventually but I have no idea what a good timeout would be. For the analyzer for example it'll scale based on codebase, on # of rules, etcetc